common/eth2/src/types.rs

```diff
@@ -1068,8 +1069,35 @@ impl FromStr for Accept {
 }
 
 fn parse_accept(accept: &str) -> Result<Vec<Mime>, String> {
-    accept
-        .split(',')
+    MediaTypeList::new(accept)
+        .into_iter()
+        .map(|item| match item {
+            Ok(MediaType {
+                ty,
+                subty,
+                suffix,
+                params,
+            }) => {
+                let suffix_format = match suffix {
+                    Some(name) => format!("+{}", name),
+                    None => String::from(""),
+                };
+
+                let _params_format: String = params
+                    .iter()
+                    .map(|(n, v)| format!("{}={}", n, v))
+                    .collect::<Vec<String>>()
+                    .join(";");
+
+                let params_format = if _params_format.is_empty() {
+                    "".to_string()
+                } else {
+                    format!(";{}", _params_format)
+                };
+                format!("{}/{}{}{}", ty, subty, suffix_format, params_format)
+            }
+            _ => "".to_string(),
```

Comment:
Rather than doing this re-serialisation in order to re-parse as a `Mime` I think we should return a vec of `MediaType` instead and refactor the usage of `parse_accept` to operate on `MediaType` rather than `Mime`.

The only place where use `parse_accept` is here, so it should be straight-forward to refactor:

https://github.com/sigp/lighthouse/blob/bd97e5f21677ca38d8074fb65b615caa75a699f8/common/eth2/src/types.rs#L1052-L1067